src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -104,7 +105,17 @@ protected virtual void Initialize()
             }
 
             string json = File.ReadAllText(hostConfigFilePath);
-            JObject hostConfig = JObject.Parse(json);
+
+            JObject hostConfig;
+            try
+            {
+                hostConfig = JObject.Parse(json);
+            }
+            catch (Exception ex)
```

Comment:
Rather than catching all exceptions here, can you just catch the specific parse exception that json.net throws?